src/ngx_http_lua_common.h

```diff
@@ -176,6 +178,9 @@ struct ngx_http_lua_main_conf_s {
 
     ngx_array_t         *preload_hooks; /* of ngx_http_lua_preload_hook_t */
 
+    ngx_array_t         *delay_list; /* of void* */
+    unsigned             delay_list_inited;
```

Comment:
@rainingmaster What do you mean by "third party module just run once in their own init phase"?